integrations/includes/Dokan/REST/VendorBogoController.php

```
@@ -110,210 +49,99 @@ public function check_permission( $request ) {
-        if ( ! $offer_vendor_id || dokan_get_current_user_id() !== $offer_vendor_id ) {
-            return new WP_REST_Response( [ 'error' => __( 'Sorry! You are not permitted to do the current action', 'storegrowth-sales-booster' ) ], 403 );
-        }
-
-        $result = $this->get_bogo()->delete( $id );
-
-        if ( is_wp_error( $result ) ) {
-            return new WP_REST_Response( [ 'error' => $result->get_error_message() ], 400 );
-        }
-
-        return new WP_REST_Response( [ 'deleted' => true ], 200 );
+    protected function check_creation_limitations( $data, $request ) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$data'. (Unused Code Rules)

(UnusedFormalParameter)